p503/arith_amd64_test.go

```diff
@@ -43,32 +73,56 @@ func testRedc(t *testing.T, redcFunc func(z *FpElement, x *FpElementX2)) {
 
 // Ensures corretness of implementation of mul operation which uses MULX
 func TestMulWithMULX(t *testing.T) {
+	defer recognizecpu()
 	if !cpu.HasBMI2 {
 		t.Skip("MULX not supported by the platform")
 	}
-	testMul(t, mulWithMULX)
+	testMul(t, kUse_MULX, kUse_MUL)
 }
 
 // Ensures corretness of implementation of mul operation which uses MULX and ADOX/ADCX
 func TestMulWithMULXADX(t *testing.T) {
+	defer recognizecpu()
 	if !(cpu.HasADX && cpu.HasBMI2) {
 		t.Skip("MULX, ADCX and ADOX not supported by the platform")
 	}
-	testMul(t, mulWithMULXADX)
+	testMul(t, kUse_MULXADX, kUse_MUL)
+}
+
+// Ensures corretness of implementation of mul operation which uses MULX and ADOX/ADCX
+func TestMulWithMULXADXAgainstMULX(t *testing.T) {
+	defer recognizecpu()
+	if !(cpu.HasADX && cpu.HasBMI2) {
+		t.Skip("MULX, ADCX and ADOX not supported by the platform")
+	}
+	testMul(t, kUse_MULX, kUse_MULXADX)
 }
 
 // Ensures corretness of Montgomery reduction implementation which uses MULX
 func TestRedcWithMULX(t *testing.T) {
+	defer recognizecpu()
 	if !cpu.HasBMI2 {
 		t.Skip("MULX not supported by the platform")
 	}
-	testRedc(t, redcWithMULX)
+	testRedc(t, kUse_MULX, kUse_MUL)
 }
 
 // Ensures corretness of Montgomery reduction implementation which uses MULX
+// and ADX
```

Comment:
We're not really being consistent with this. But BMI2 is the instruction set extension that adds the MULX instruction. And ADX is the instruction set extension that contains the ADCX and ADOX instructions.  Here it's about the instructions, so it should say ADOX/ADCX.

This is also a bit confusing in `cpuid.go` and in  a number of labels.